### PR TITLE
PRO-263 paginate list of offices in selectofficemodal

### PR DESCRIPTION
**Pull request by yuuks**
These changes incorporate the `Paginator` component into `SelectOfficeModal`

Changes
---
* Makes [SearchData](https://github.com/ProjectProtocol/project-protocol-web/pull/100/files#diff-535158f23577b58f727d33d45f1eea7ed731e43f9ca0d902b1a48892c912ecdfR6) utilize generics for the data type it contains.
* Uses the new [offices endpoint](https://github.com/ProjectProtocol/project-protocol-web/pull/100/files#diff-04eeaf5717a7caf285cf1258c24a8253b79c74bb0d326ec336b2dd8cb16109dcR17) for fetching the list of offices.
* Uses the Paginator component to fetch the offices in pages.

Testing
---
* In the deploy preview, go to the Add New Agent view, and click on `select an office`. Scroll through and try searching.

### CODE REVIEW

**Comment by yuuks** [comment]
Overall, looking for feedback on patterns and structure, which I'm still getting a feel for.

In addition there's an issue where when you update the search text, it will show the previous result's search for a sec. (for example: type 'a', wait for results to display, then delete and type 'l', and you'll see the previous results displayed until the results are fetched). So need to figure out a way to clear results when search text changes.

File: src/api/search.ts [hunk 1]
```diff
@@ -3,11 +3,13 @@ import Office from '../types/Office'
 import SearchMeta from '../types/SearchMeta'
 import apiClient from './client'
 
-export type SearchData = {
+export type SearchData<T> = {
```
Comment (tasandberg) [inline src/api/search.ts:6]: I think this SearchData should be deleted in favor of the new `SearchData.ts` definition in `src/types`

**Review by tasandberg — CHANGES_REQUESTED** [review]
Code looks good and list loads great, tested in preview and local. Only nit was to remove the old SearchData type declaration in `src/api/search.ts.`

**Review by tasandberg — APPROVED** [review]

**Comment by tasandberg** [comment]
@yuuks In the future feel free to merge branches once approved. I'm taking the liberties since I'm pairing with Eddie and we need the new SearchData type :D

**Pull request merged**